### ISSUE TITLE
Conflict with type syntax

### ISSUE THREAD

**Issue by sebmck** [zenparsing/es-typed-catch#1]
TypeScript and Flow both allow type annotations on catch parameters like so:

``` javascript
try {
  throw "foobar";
} catch (err: string) {}
```

This will conflict as the parameter will not always reference a binding and the semantics here may not be warranted which introduces a refactoring hazard.

This wouldn't really be an issue since I don't think those projects should effect how JavaScript implements something but I was under the assumption that the colon syntax was reserved? ([TC39 notes](https://github.com/rwaldron/tc39-notes/blob/244797871dfa8fd475fd593821605fee14d2cb05/es6/2014-09/sept-25.md#types)) Although I'm not sure if there was follow up work to be done or if this decision was reversed?

**Comment by sebmck** [comment]
Also I know this repo is only an hour old so apologies if this feedback is premature. :stuck_out_tongue: Thanks for putting it together!

**Comment by zenparsing** [comment]
This is definitely not ready for public consumption and probably about to fall over at any moment : )

I was trying to find out what TS and Flow do with catch clauses, but I couldn't find any good information.  When I tried out both the TS and Flow playgrounds, I got syntax errors if I put a type annotation into the catch clause.  Am I doing something wrong?

In any case, I ultimately would like it to be compatible (more or less?) with type annotation syntax, since it's basically a dynamic type check.  I also don't know if the semantics of the dynamic type check are right yet...
